src/zcash/JoinSplit.cpp

```diff
@@ -345,8 +293,11 @@ class JoinSplitCircuit : public JoinSplit<NumInputs, NumOutputs> {
         // estimate that it doesn't matter if we check every time.
         pb.constraint_system.swap_AB_if_beneficial();
 
+        r1cs_ppzksnark_proving_key<ppzksnark_ppT> pk;
+        loadFromFile(pkPath, pk);
+
         return ZCProof(r1cs_ppzksnark_prover<ppzksnark_ppT>(
-            *pk,
+            pk,
```

Comment:
Was removing the use of Boost::optional something you intended to do anyways? Or is it related to the streaming change?